src/styles/blocks/calendar.scss

```
@@ -0,0 +1,54 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $blockGap;
+
+  padding: $blockPadding;
+  max-width: getWidth($blockSize, $blockGap, $blockPadding);
+
+  @each $day, $position in $week {
+    &--start-day_#{$day} :first-child {
+      margin-left: ($blockSize + $blockGap) * $position;
+    }
+  }
+
+  @for $index from 1 through 31 {
+    &--month-length_#{$index} :nth-child(n + #{$index + 1}) {
+      display: none;
+    }
+  }
+
+  &__day {
+    @include daySize($blockSize);
+
+    display: flex;
+    background-color: #eee;
+    border: 1px solid #000;
+  
+    font-family: Arial, Helvetica, sans-serif;
+    font-size: 30px;
+    cursor: pointer;
+
+    transition-duration: 0.5s;
```

Always list styles that you apply transition to:
```suggestion
    transition-duration: background-color 0.5s, transform 0.5s;
```